spec/models/rook_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe Rook, type: :model do
+  describe "#valid_move?" do
+    let(:game) {create :game}
+
+    it "should properly evaluate malformed moves that are otherwise valid" do
+      white_rook = game.pieces.find_by(rank: 1, file: 1)
+      black_rook = game.pieces.find_by(rank: 8, file: 1)
+      expect(white_rook.valid_move?(0, 1)).to eq(false)
+      expect(white_rook.valid_move?(1, 0)).to eq(false)
+      expect(white_rook.valid_move?(0, 0)).to eq(false)
+      expect(white_rook.valid_move?(0, 2)).to eq(false)
+      expect(white_rook.valid_move?(2, 0)).to eq(false)
+      expect(white_rook.valid_move?(1, 1)).to eq(false)
+      expect(black_rook.valid_move?(8, 1)).to eq(false)
+      expect(black_rook.valid_move?(7, 0)).to eq(false)
+      expect(black_rook.valid_move?(8, 0)).to eq(false)
+      expect(black_rook.valid_move?(9, 0)).to eq(false)
+      expect(black_rook.valid_move?(9, 1)).to eq(false)
+      expect(black_rook.valid_move?(9, 2)).to eq(false)
+    end
+
+    it "should properly evaluate obstructed moves that are otherwise valid" do
+      white_rook = game.pieces.find_by(rank: 1, file: 1)
+      black_rook = game.pieces.find_by(rank: 8, file: 1)
+      expect(white_rook.valid_move?(1, 2)).to eq(false)
+      expect(white_rook.valid_move?(2, 1)).to eq(false)
+      expect(black_rook.valid_move?(8, 2)).to eq(false)
+      expect(black_rook.valid_move?(7, 1)).to eq(false)
+    end
```

Comment:
Right here, could you also cover an obstructed horizontal move over to a lower file? I would take care of this by grabbing the black rook at h8, instead of the one at a8.